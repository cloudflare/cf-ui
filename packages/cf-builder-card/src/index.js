import cardActions from './cardActions';
import CardActionTypes from './CardActionTypes';
import CardBuilder from './CardBuilder';
import cardReducer from './cardReducer';

export { cardActions, CardActionTypes, CardBuilder, cardReducer };
