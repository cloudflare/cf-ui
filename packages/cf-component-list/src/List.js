import React from 'react';

import PropTypes from 'prop-types';

class List extends React.Component {
  render() {
    let tagName = 'ul';
    let className = 'cf-list';

    if (this.props.ordered) {
      tagName = 'ol';
      className += ' cf-list--ordered';
    }

    if (this.props.unstyled) {
      className += ' cf-list--unstyled';
    }

    return React.createElement(tagName, { className }, this.props.children);
  }
}

List.propTypes = {
  ordered: PropTypes.bool,
  unstyled: PropTypes.bool,
  children: PropTypes.node
};

List.defaultProps = {
  ordered: false,
  unstyled: false
};

export default List;
