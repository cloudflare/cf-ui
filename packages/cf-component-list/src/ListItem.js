import React from 'react';

import PropTypes from 'prop-types';

class ListItem extends React.Component {
  render() {
    return (
      <li className="cf-list__item">
        {this.props.children}
      </li>
    );
  }
}

ListItem.propTypes = {
  children: PropTypes.node
};

export default ListItem;
