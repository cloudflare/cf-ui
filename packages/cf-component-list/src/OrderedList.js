import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  listStyleType: theme.listStyleType,
  listStylePosition: theme.listStylePosition,
  marginLeft: theme.marginLeft
});

const OrderedList = ({ children, className }) => (
  <ol className={className}>{children}</ol>
);

OrderedList.propTypes = {
  children: PropTypes.node
};

export default createComponent(styles, OrderedList);
