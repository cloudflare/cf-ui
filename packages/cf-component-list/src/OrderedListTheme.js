export default baseTheme => ({
  listStyleType: 'decimal',
  listStylePosition: 'outside',
  marginLeft: '3em'
});
