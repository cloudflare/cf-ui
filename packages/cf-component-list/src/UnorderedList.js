import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  listStyleType: theme.listStyleType,
  listStylePosition: theme.listStylePosition,
  marginLeft: theme.marginLeft
});

const UnorderedList = ({ children, className }) => (
  <ul className={className}>{children}</ul>
);

UnorderedList.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

export default createComponent(styles, UnorderedList);
