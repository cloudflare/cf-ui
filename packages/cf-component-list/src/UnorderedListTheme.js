export default baseTheme => ({
  listStyleType: 'disc',
  listStylePosition: 'outside',
  marginLeft: '3em'
});
