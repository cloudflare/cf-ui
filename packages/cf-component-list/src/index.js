import { applyTheme } from 'cf-style-container';

import UnorderedListUnstyled from './UnorderedList';
import UnorderedListTheme from './UnorderedListTheme';
import OrderedListUnstyled from './OrderedList';
import OrderedListTheme from './OrderedListTheme';

import ListItemUnstyled from './ListItem';
import ListItemTheme from './ListItemTheme';

const UnorderedList = applyTheme(UnorderedListUnstyled, UnorderedListTheme);
const OrderedList = applyTheme(OrderedListUnstyled, OrderedListTheme);
const ListItem = applyTheme(ListItemUnstyled, ListItemTheme);

export {
  UnorderedList,
  UnorderedListUnstyled,
  UnorderedListTheme,
  OrderedList,
  OrderedListUnstyled,
  OrderedListTheme,
  ListItem,
  ListItemUnstyled,
  ListItemTheme
};
