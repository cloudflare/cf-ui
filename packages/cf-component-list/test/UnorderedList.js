import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { UnorderedList } from '../../cf-component-list/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(<UnorderedList>List</UnorderedList>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
