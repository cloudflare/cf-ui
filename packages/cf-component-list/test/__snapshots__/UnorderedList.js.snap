// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<ul
  className="a b c"
>
  List
</ul>
`;

exports[`should render 2`] = `
"
.a {
  list-style-type: disc
}

.b {
  list-style-position: outside
}

.c {
  margin-left: 3em
}
"
`;
